### ISSUE TITLE
[Bug]: Text Input (Content Tab) - Limit option (Off by one?)

### ISSUE THREAD

**Issue by usbrandon** [apache/hop#2886]
### Apache Hop version?

2.4-GA

### Java version?

11

### Operating system

Linux

### What happened?

I wanted only to output the first two rows in my CSV file.  When I put 2 in the box, only 1 row came out. When I entered 3, the 2 rows I desired came out.  That feels wrong.  Is this a bug?

![image](https://user-images.githubusercontent.com/1233654/235231367-d97d4b3f-9519-48c1-8f84-75568cf5ff69.png)

### Issue Priority

Priority: 3

### Issue Component

Component: Transforms

**Comment by hansva** [comment]
My guess is that the header is also counted as a written line, so you end up with header (your header is set to 1 line) + 2 data rows if you put limit 3

**Comment by bamaer** [comment]
made the same assumption, which seems to be [correct](https://github.com/apache/hop/blob/81c3484bf69bdfb867dd76a3013ed5194f4497b8/plugins/transforms/textfile/src/main/java/org/apache/hop/pipeline/transforms/fileinput/text/TextFileInputReader.java#L398)
backward compatibility issues are probably extremely limited in number, but not 0, so the safest approach seems to me to update the description and docs

**Comment by usbrandon** [comment]
Yes, I would agree with updating any hint or description near that label.  It is not logical to even consider it the way that it is.  It is an "Input" and cannot even possibly output a "header row" as part of the normal output that one would consider limiting in the first place.  So now that it is baked in, we have to prevent people from making mistakes.

**Comment by bjackson-ep** [comment]
What if we could signal to the user that they should upgrade options in a transform and record in the pipeline some version number where we know certain things are improved?  Or... what if we have a pipeline / workflow scanner that could be a "are you ready for an upgrade?" tool and let it optionally update things like this.  You know it would only be something like -1 to what they had.

**Comment by mattcasters** [comment]
Please re-open if there is consensus on the desired outcome.
